packages/framework/src/validators/base.validator.ts

```diff
@@ -14,20 +21,31 @@ export const validateData = async <
   schema: T_Schema,
   data: T_Unvalidated
 ): Promise<ValidateResult<T_Validated>> => {
-  if (zodValidator.canHandle(schema)) {
-    return zodValidator.validate(data, schema);
-  } else if (jsonSchemaValidator.canHandle(schema)) {
-    return jsonSchemaValidator.validate(data, schema);
+  /**
+   * TODO: Replace type coercion with async type guard when available.
+   *
+   * @see https://github.com/microsoft/typescript/issues/37681
+   */
+  if (await zodValidator.canHandle(schema)) {
```

Comment:
`canValidated` now returns a `Promise` to account for the `checkDependencies` usage of `await import(...)`